Source/ARTRestChannel.m

```diff
@@ -229,104 +230,94 @@ - (void)internalPostMessages:(id)data callback:(void (^)(ARTErrorInfo *__art_nul
+                    NSData *baseIdData = [ARTCrypto generateSecureRandomData:kIdempotentLibraryGeneratedIdLength];
+                    baseId = [baseIdData base64EncodedStringWithOptions:0];
+                }
             }
-            else {
-                errorInfo = error ? [ARTErrorInfo createFromNSError:error] : nil;
+            
+            NSInteger serial = 0;
+            for (ARTMessage *message in messages) {
+                if (message.clientId && self.rest.auth.clientId_nosync && ![message.clientId isEqualToString:self.rest.auth.clientId_nosync]) {
+                    callback([ARTErrorInfo createWithCode:ARTStateMismatchedClientId message:@"attempted to publish message with an invalid clientId"]);
+                    return;
+                }
+                if (baseId) {
+                    message.id = [NSString stringWithFormat:@"%@:%ld", baseId, (long)serial];
+                }
+                serial += 1;
+            }
+            
+            NSError *encodeError = nil;
+            encodedMessage = [self.rest.defaultEncoder encodeMessages:data error:&encodeError];
+            if (encodeError) {
+                callback([ARTErrorInfo createFromNSError:encodeError]);
+                return;
             }
-            callback(errorInfo);
         }
-    }];
-});
+        
+        NSURLComponents *components = [[NSURLComponents alloc] initWithURL:[NSURL URLWithString:[self->_basePath stringByAppendingPathComponent:@"messages"]] resolvingAgainstBaseURL:YES];
+        NSMutableArray<NSURLQueryItem *> *queryItems = [NSMutableArray new];
+        
+        if (queryItems.count > 0) {
+            components.queryItems = queryItems;
+        }
+        
+        NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:[components URL]];
+        request.HTTPMethod = @"POST";
+        request.HTTPBody = encodedMessage;
+        
+        if (self.rest.defaultEncoding) {
+            [request setValue:self.rest.defaultEncoding forHTTPHeaderField:@"Content-Type"];
+        }
+        
+        [self.logger debug:__FILE__ line:__LINE__ message:@"RS:%p C:%p (%@) post message %@", self->_rest, self, self.name, [[NSString alloc] initWithData:encodedMessage encoding:NSUTF8StringEncoding]];
+        
+        [self->_rest executeRequest:request withAuthOption:ARTAuthenticationOn completion:^(NSHTTPURLResponse *response, NSData *data, NSError *error) {
+            if (callback) {
+                ARTErrorInfo *errorInfo;
+                if (self->_rest.options.addRequestIds) {
+                    errorInfo = error ? [ARTErrorInfo wrap:[ARTErrorInfo createFromNSError:error] prepend:[NSString stringWithFormat:@"Request '%@' failed with ", request.URL]] : nil;
+                } else {
+                    errorInfo = error ? [ARTErrorInfo createFromNSError:error] : nil;
+                }
+                
+                callback(errorInfo);
+            }
+        }];
+    });
 }
 
```

Comment:
Looks heavily rewritten. I guess you've just moved setting `request_id` to ARTRest.m?

Reply:
nothing but the indentaion changed here

Reply:
I mean all changes in this commit for this file.